test/integration/integration.h

```diff
@@ -213,6 +213,8 @@ class BaseIntegrationTest : Logger::Loggable<Logger::Id::testing> {
     xds_stream_->sendGrpcMessage(discovery_response);
   }
 
+  size_t ClusterMemoryHelper(int num_clusters, bool allow_stats);
```

Comment:
Looks good conceptually, but you can throw this into a new subclass of BaseIntegrationTest local to stats_integration_test.cc, rather than modifying the common base class? Maybe call it ClusterMemoryUtilizationTestHelper or similar?

Or, if you think this is generally useful to use beyond the stats integration test (which it very well might be) you could put a doxygen comment here. It looks specific to stats though.

Reply:
I think there's a potential for it to be useful beyond stats if we want to configure varying filters or see how a particular feature behaves with a large workload. I'm fine trying to subclass if we don't think it will be useful.

Reply:
Looking at this again it does make sense to just keep it in stats. I'll move it back there now.